forte/processors/data_augment/algorithms/distribution_replacement_augmenter.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from typing import Dict, List
+
+import nltk
+from nltk.corpus import words, stopwords
+from ft.onto.base_ontology import Token
+
+from forte.processors.data_augment.algorithms.base_augmenter \
+    import ReplacementDataAugmenter
+
+
+__all__ = [
+    "DistributionReplacementAugmenter",
+]
+
+
+class DistributionReplacementAugmenter(ReplacementDataAugmenter):
+    r"""
+    This class is a data augmenter utilizing the NLTK vocabulary,
+    from which sample by a given distribution to subsitute a word
+    from the original text. The distributions implemented now are uniform,
+    unigram (related to a specific dataset).
+    Todo: More distriubtion types can be added.
+    """
+    def __init__(self, configs: Dict[str, str]):
+        random.seed(0)
+        super().__init__(configs)
+        try:
+            self.vocab = set(words.words())
+            self.stop_words = set(stopwords.words(self.configs['lang']))
+        except LookupError:
+            nltk.download('words')
+            nltk.download('stopwords')
+            self.vocab = set(words.words('en'))
+            self.stop_words = set(stopwords.words(self.configs['lang']))
+
+    @property
+    def replacement_level(self) -> List[str]:
+        return ["word"]
+
+    def _uniform_sample(self) -> str:
```

Comment:
And also make these sample methods into individual classes so that we can pass them in.